### DISCUSSION TITLE
How to use Redux properly in latest NextJs

### DISCUSSION THREAD

**Discussion by Marcel-Konjata** [vercel/next.js#17390] [Help] [answered]
Hi, I am working with next a little while and I am looking at examples with redux at main canary/examples Github repo of Next.
Tbh I don't really know if it is desired to have 2 stores,  and adds only more confusion. To me, the issue is if you have declared hydration on getStatic/serverSide props on each change of route you will override the client store right, that means that you always lose progress made in a different route, or there is some approach that will help you preserve progress and "throw-away server store" if you made some update or progress on client?.

Clearly, I would need some help to understand how to use Redux in Next Js correctly, if there is some guide article for more "dummies" because this is only one thing that I don't really get. Redux-wrapper seems not to fill that functionality or at least it exactly does what I have described above.

**Comment by lfades** [comment, marked as answer]
There is only one store (the one in the browser), the server creates a store (that's only there for pre-rendering) to make sure that build-time pre-rendering (getStaticProps) and runtime pre-rendering (getServerSideProps) works properly. The state is returned and props and then hydrated by the browser store. It's not an optional process as you can't (and should not) send the server store to the browser.

How to update the browser store is up to you, the examples don't go deep into complex stores.
